icu4c/source/test/intltest/rbbitst.cpp

```diff
@@ -2606,6 +2607,143 @@ RBBISentMonkey::~RBBISentMonkey() {
 
 
 
+class SegmentationRule {
+  public:
+    enum Resolution {
+        BREAK = u'÷',
+        NO_BREAK = u'×',
+    };
+    struct BreakContext {
+        BreakContext(std::size_t index) : indexInRemapped(index) {}
+        std::optional<std::size_t> indexInRemapped;
+        const SegmentationRule *appliedRule = nullptr;
+    };
+
+    SegmentationRule(std::u16string_view name)
+        : name_(UnicodeString(name).toUTF8String(std::string{})) {}
+    virtual ~SegmentationRule() = default;
+
+    virtual void apply(UnicodeString &remapped, std::vector<BreakContext> &resolved) const = 0;
+    virtual Resolution resolution() const = 0;
+    const std::string &name() const { return name_; }
+
+  private:
+    const std::string name_;
+};
+
+class RemapRule : public SegmentationRule {
+  public:
+    RemapRule(const std::u16string_view name, const std::u16string_view pattern,
+              const std::u16string_view replacement)
+        : SegmentationRule(name), replacement_(replacement) {
+        UParseError parseError;
+        UErrorCode status;
+        pattern_.reset(RegexPattern::compile(pattern, parseError, status));
+        U_ASSERT(U_SUCCESS(status));
```

Comment:
Ah, I did not know about IcuTestErrorCode. I will look into that.

Reply:
I ended up just `std::terminate`ing with a message as I have done elsewhere, since bubbling up the error all the way to the test would make debugging tricky.